interpolator/include/interpolator.inc

```diff
@@ -472,59 +521,51 @@ if(dimension_exists(fileobj, "time")) then
            !    include the base_time; values will be generated relative to the
            !    "real" time.
            !--------------------------------------------------------------------
-           if ( (model_calendar == JULIAN .and.   &
-                & trim(adjustl(lowercase(file_calendar))) == 'julian')  .or. &
-                & (model_calendar == NOLEAP .and.   &
-                & trim(adjustl(lowercase(file_calendar))) == 'noleap') )  then
+           if ( (model_calendar == JULIAN .and. trim(adjustl(lowercase(file_calendar))) == 'julian')  .or. &
+                & (model_calendar == NOLEAP .and. trim(adjustl(lowercase(file_calendar))) == 'noleap') )  then
 
              !---------------------------------------------------------------------
              !    no calendar conversion needed.
              !---------------------------------------------------------------------
              clim_type%time_slice(n) = &
-               set_time( INT( (time_in(n)-real(INT(time_in(n)),FMS_INTP_KIND_)) &
-                               *real(SECONDS_PER_DAY,FMS_INTP_KIND_)), INT(time_in(n))) &
-                 + base_time
+               set_time( INT( (time_in(n)-real(INT(time_in(n)),r8_kind))*SECONDS_PER_DAY), &
+               INT(time_in(n))) + base_time
 
            !---------------------------------------------------------------------
            !    convert file times from noleap to julian.
            !---------------------------------------------------------------------
-           else if ( (model_calendar == JULIAN .and.   &
-                & trim(adjustl(lowercase(file_calendar))) == 'noleap')) then
-             Noleap_time = set_time (0, INT(time_in(n))) + base_time
-             call get_date_no_leap (Noleap_time, yr, mo, dy, hr,  &
-                                    mn, sc)
-             clim_type%time_slice(n) = set_date_julian (yr, mo, dy,  &
-                                                        hr, mn, sc)
+           else if ( (model_calendar == JULIAN .and. trim(adjustl(lowercase(file_calendar))) == 'noleap')) then
+              Noleap_time = set_time( INT((time_in(n)-real(INT(time_in(n)),r8_kind))*SECONDS_PER_DAY), &
```

Comment:
Changed to also take into account fractional number of days (converted to seconds) when setting time for greater accuracy.  The same is done for the case where the model calendar and the file calendar are the same (see above).